tests/AsyncClientTest.php

```diff
@@ -35,6 +35,8 @@ public function testInLoop()
             } catch (\Throwable $e) {
                 $exception = $e;
             }
+
+            Loop::stop();
```

Comment:
You can just use `wait` instead of the yield and remove the `Loop::run()` and `Loop::stop()`.

Reply:
That's the main goal, to test the promise / stream inside an existing loop, with the `ResponseStream` this code was failing

Reply:
Ah, correct, because `(string) $response->getBody();` did / does an internal `wait`. Maybe add a comment then?

Reply:
Done, is there anything left or are we good ? (can squash it if you want)